src/lib/utils.ts

```diff
@@ -87,32 +84,98 @@ export const promiseErrorMultiplier = <T, Q extends Error>(
   });
 };
 
+export const generateGatewayMap = async (
+  l2Multicaller: MultiCaller,
+  l2Network: L2Network,
+  l1Provider: Provider
+) => {
+  const fromBlock = 0;
+  const toBlock = await l1Provider.getBlockNumber();
+  const topic0 = ethers.utils.id('GatewaySet(address,address)');
+  const l1GatewayResults: Map<string, string> = new Map();
```

Comment:
ooo hmm another fun edge case is if two gateways get deployed across two different networks with the same L1 address 😰

Reply:
You mean one L1 gateway mapped by 2 network? I thought if arbitrum one has gateway A and nova has gateway B (they all mapping to same gateway on L1), but L1 only mapped to A, so B will be invalid?

Reply:
Also, we get l1's gateway address from gateway router, and gateway router is specific for each l2 network.

Reply:
woops, you're right I'm wrong, I had the wrong mapping in my head, plz disregard